features/steps/dataset_feature.go

```diff
@@ -0,0 +1,228 @@
+package feature
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+	"time"
+
+	"github.com/ONSdigital/dp-dataset-api/config"
+	"github.com/ONSdigital/dp-dataset-api/models"
+	"github.com/ONSdigital/dp-dataset-api/mongo"
+	"github.com/ONSdigital/dp-dataset-api/service"
+	serviceMock "github.com/ONSdigital/dp-dataset-api/service/mock"
+	"github.com/ONSdigital/dp-dataset-api/store"
+	storeMock "github.com/ONSdigital/dp-dataset-api/store/datastoretest"
+	"github.com/ONSdigital/dp-healthcheck/healthcheck"
+	kafka "github.com/ONSdigital/dp-kafka/v2"
+	"github.com/ONSdigital/dp-kafka/v2/kafkatest"
+	featuretest "github.com/armakuni/dp-go-featuretest"
+	"github.com/benweissmann/memongo"
```

Comment:
while the memongo module works, have concerns that it is not in high use, unlike the node counterpart. Also that that it needs to download a copy of mongo on every new container run.

Reply:
In CI, you could cache the vendor libraries in the task